Tasks/AzureMonitorAlerts/azurerestutility.ts

```diff
@@ -0,0 +1,265 @@
+    HttpClient, 
+    HttpClientResponse 
+} from "typed-rest-client/HttpClient";
+
+import { 
+    RestClient, 
+    IRequestOptions, 
+    IRestResponse 
+} from "typed-rest-client/RestClient";
+
+import * as Util from "./utility";
+import { 
+    IAzureMetricAlertRequestBody,
+    IAzureMetricAlertRule
+} from "./interfaces";
+
+var httpObj = new HttpClient(tl.getVariable("AZURE_HTTP_USER_AGENT"));
+var restObj = new RestClient(tl.getVariable("AZURE_HTTP_USER_AGENT"));
+
+var defaultAuthUrl = 'https://login.windows.net/';
+
+function getAuthorizationToken(endPoint): Q.Promise<string> {
+
+    let deferred: Q.Deferred<string> = Q.defer<string>();
+    let envAuthUrl = (endPoint.envAuthUrl) ? (endPoint.envAuthUrl) : defaultAuthUrl;
+    let authorityUrl = envAuthUrl + endPoint.tenantID + "/oauth2/token/";
+    let requestData = querystring.stringify({
+        resource: endPoint.activeDirectoryResourceId,
+        client_id: endPoint.servicePrincipalClientID,
+        grant_type: "client_credentials",
+        client_secret: endPoint.servicePrincipalKey
+    });
+
+    let requestHeader = {
+        "Content-Type": "application/x-www-form-urlencoded; charset=utf-8"
+    }
+
+    tl.debug('Requesting for Auth Token: ' + authorityUrl);
+    httpObj.post(authorityUrl, requestData, requestHeader)
+        .then(async (response: HttpClientResponse) => {
+            if (response.message.statusCode == 200) {
+                let contents: string = await response.readBody();
+                if(!!contents) {
+                    deferred.resolve(JSON.parse(contents).access_token);
+                }
+            }
+            else {
+                deferred.reject(tl.loc('CouldnotfetchaccesstokenforAzureStatusCode', response.message.statusCode, response.message.statusMessage));
+            }
+        }, (error) => {
+            deferred.reject(error);
+        });
+
+    return deferred.promise;
+}
+
+export async function getAzureRmResourceDetails(
+    endpoint, 
+    resourceUri: string) : Promise<any> {
+    
+    let deferred: Q.Deferred<any> = Q.defer<any>();
+    let accessToken: string = await getAuthorizationToken(endpoint);
+   
+    // resourceUri is of the form : /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}
+
+    let splittedResourceUri: string[] = resourceUri.split("/");
+    let resourceGroupName: string = splittedResourceUri[4];
+    let resourceType: string = splittedResourceUri[6] + "/" +  splittedResourceUri[7];
+    let resourceName: string = splittedResourceUri[8];
+
+    tl.debug(`Getting AzureRm resource details - '${resourceName}' in resource group '${resourceGroupName}'`);
+
+    let apiVersion = "2017-05-10";
+    let restUrl = `${endpoint.url}subscriptions/${endpoint.subscriptionId}/resourceGroups/${resourceGroupName}/resources?$filter=resourceType EQ '${resourceType}' AND name EQ '${resourceName}'&api-version=${apiVersion}`;
```

Comment:
the url's being created almost always have a common prefix. should this be abstracted out of each function and made into a separate url generator function?